### PR TITLE
Questions

### PR DESCRIPTION
**Pull request by TheJarles**
Same as the earlier pull request ([https://github.com/kolibre/kobra/pull/4](url)) but amended with a bug fix and removed a merge conflict I had forgotten.

### CODE REVIEW

File: app/models/user_content.rb [hunk 1]
```diff
@@ -2,6 +2,10 @@ class UserContent < ApplicationRecord
   belongs_to :user
   belongs_to :content
   belongs_to :content_list
+<<<<<<< HEAD
```
Comment (jabbors) [inline app/models/user_content.rb]: This change shall be reverted of fixed. Originates from an unresolved merge conflict.

File: db/migrate/20190221181952_create_question_texts.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateQuestionTexts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :question_texts do |t|
+      t.references :language, foreign_key: true
+      t.integer :text, null: false
```
Comment (jabbors) [inline db/migrate/20190221181952_create_question_texts.rb]: Bad type in the reference database, shall be `text`.

File: db/schema.rb [hunk 3]
```diff
@@ -72,6 +100,59 @@
...
+  create_table "question_texts", force: :cascade do |t|
+    t.integer "language_id"
+    t.integer "text", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["language_id"], name: "index_question_texts_on_language_id"
+  end
+
+  create_table "question_types", force: :cascade do |t|
+    t.string "type_name", null: false
```
Comment (jabbors) [inline db/schema.rb]: Let's simplify the column name to `name`. It's implied from the table name that it's a type of some kind.

**Comment by TheJarles** [comment]
I implemented the requested changes. I also took the liberty to change some of the types of the columns in QuestionInput (text_numeric and text_alphanumeric => text instead of integer, and default value for allow_multiple_selections from nil to 0 as it is a integer-represented boolean, correct?). I also noticed that the whitelisted variables names in the  models' respective controllers were not updated after initially generated, so I updated those as well.

File: db/migrate/20190221181516_create_question_inputs.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class CreateQuestionInputs < ActiveRecord::Migration[5.2]
+  def change
+    create_table :question_inputs do |t|
+      t.references :question, foreign_key: true
+      t.integer :allow_multiple_selections, default: 0
+      t.text :text_numeric, default: nil
```
Comment (jabbors) [inline db/migrate/20190221181516_create_question_inputs.rb]: `text_numeric` and `text_alphanumeric` are not used to store input from the user, the purpose is instead to define which input types are supported for a question sent from the server to the user.

I checked the code and we only check if the value is null or not when we build the input questions. Thus it does not really matter of which type they are but to be consistent and save a few bytes we shall change them back to integers.

Comment (TheJarles) [reply db/migrate/20190221181516_create_question_inputs.rb]: I reverted the type changes, so now I just need a new approval for the pull request.

Comment (jabbors) [reply db/migrate/20190221181516_create_question_inputs.rb]: You did not actually revert anything, instead you change another column from integer to text 😄 

So now you have two reverts to do 😄 In this table, fields `text_numeric`, `text_alphanumeric` and `audio` should be of type integer. Not that it really matters be we save a few bytes and that's an optimisation that is better to do know than later.

Comment (TheJarles) [reply db/migrate/20190221181516_create_question_inputs.rb]: Hahah, that's what happens when you're unfocused. I'll fix it in a bit.

**Pull request merged**